ArtemisKit/Sources/Messages/Services/MessagesService/MessagesServiceImpl.swift

```diff
@@ -258,6 +258,33 @@ struct MessagesServiceImpl: MessagesService {
         }
     }
 
+    struct UploadFileResult: Codable {
+           let path: String
+       }
+
+       func uploadFile(for courseId: Int, and conversationId: Int64, file: Data, filename: String, mimeType: String) async -> DataState<String> {
```

Comment:
Could we remove `uploadImage` and replace its usages with a call to this function? From what I can see it's the same, just with images using someUUID.jpg as fileName and Image/jpeg as mimeType.